examples/mir_demo_server/server_example.cpp

```diff
@@ -118,283 +119,25 @@ catch (std::exception const& error)
-        }
-    });
-}
-
-void DemoConfigFile::add_bool_attribute(live_config::Key const& key, std::string_view description, HandleBool handler)
-{
-    add_string_attribute(key, description, [handler](live_config::Key const& key, std::optional<std::string_view> val)
-    {
-        if (val)
-        {
-            if (*val == "true")
-            {
-                handler(key, true);
-            }
-            else if (*val == "false")
-            {
-                handler(key, false);
-            }
-            else
-            {
-                mir::log_warning(
-                    "Config key '%s' has invalid integer value: %s",
-                    key.to_string().c_str(),
-                    std::format("{}",*val).c_str());
-                handler(key, std::nullopt);
-            }
-        }
-        else
-        {
-            handler(key, std::nullopt);
-        }
-    });
-}
-
-void DemoConfigFile::add_float_attribute(live_config::Key const& key, std::string_view description, HandleFloat handler)
-{
-    add_string_attribute(key, description, [handler](live_config::Key const& key, std::optional<std::string_view> val)
-    {
-        if (val)
-        {
-            float parsed_val = 0;
-
-            auto const [_, err] = std::from_chars(val->data(), val->data() + val->size(), parsed_val);
-
-            if (err == std::errc{})
-            {
-                handler(key, parsed_val);
-            }
-            else
-            {
-                mir::log_warning(
-                    "Config key '%s' has invalid integer value: %s",
-                    key.to_string().c_str(),
-                    std::format("{}",*val).c_str());
-                handler(key, std::nullopt);
-            }
-        }
-        else
-        {
-            handler(key, std::nullopt);
-        }
-    });
-}
-
-void DemoConfigFile::add_string_attribute(live_config::Key const& key, std::string_view, HandleString handler)
-{
-    std::lock_guard lock{config_mutex};
-    attribute_handlers[key] = handler;
-}
 }
 
 int main(int argc, char const* argv[])
 try
 {
     miral::MirRunner runner{argc, argv, "mir/mir_demo_server.config"};
 
-    DemoConfigFile demo_configuration{runner, "mir_demo_server.live-config"};
+    miral::live_config::IniFile config_store;
     runner.set_exception_handler(exception_handler);
 
-    miral::CursorScale cursor_scale{demo_configuration};
-    miral::OutputFilter output_filter{demo_configuration};
-    miral::InputConfiguration input_configuration{demo_configuration};
-    demo_configuration.handle_initial_config();
+    miral::CursorScale cursor_scale{config_store};
+    miral::OutputFilter output_filter{config_store};
+    miral::InputConfiguration input_configuration{config_store};
+
+    miral::ConfigFile config_file{
+        runner,
+        "mir_demo_server.live-config",
+        miral::ConfigFile::Mode::reload_on_change,
+        [&config_store](auto&... args){ config_store.load_file(args...); }};
```

Comment:
Is this too big an incantation for user code? Would a helper function make sense? Or another `ConfigFile` constructor?